test/bigtable/table.js

```diff
@@ -0,0 +1,852 @@
+      var error = new Error('err');
+
+      beforeEach(function() {
+        table.requestStream = function() {
+          var stream = new Stream({
+            objectMode: true
+          });
+
+          setImmediate(function() {
+            stream.emit('error', error);
+          });
+
+          return stream;
+        };
+      });
+
+      it('should emit an error event', function(done) {
+        table.getRows()
+          .on('error', function(err) {
+            assert.strictEqual(error, err);
+            done();
+          })
+          .on('data', done);
+      });
+
+      it('should return an error to the callback', function(done) {
+        table.getRows(function(err, rows) {
+          assert.strictEqual(error, err);
+          assert(!rows);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('insert', function() {
+    it('should create an "insert" mutation', function(done) {
+      var entries = [{
+        key: 'a',
+        data: {}
+      }, {
+        key: 'b',
+        data: {}
+      }];
+
+      table.mutate = function(entries) {
+        assert.deepEqual(entries[0], {
+          key: entries[0].key,
+          data: entries[0].data,
+          method: 'insert'
+        });
+
+        assert.deepEqual(entries[1], {
+          key: entries[1].key,
+          data: entries[1].data,
+          method: 'insert'
+        });
+
+        done();
+      };
+
+      table.insert(entries, assert.ifError);
+    });
+  });
+
+  describe('mutate', function() {
+    it('should provide the proper request options', function(done) {
+      var entries = [{}, {}];
+
+      table.request = function(grpcOpts, reqOpts) {
+        assert.deepEqual(grpcOpts, {
+          service: 'BigtableService',
+          method: 'mutateRows'
+        });
+
+        assert(reqOpts.tableName, TABLE_NAME);
+        assert(reqOpts.entries, entries);
+
+        var parseSpy = FakeMutation.parse;
+
+        assert.strictEqual(parseSpy.callCount, 2);
+        assert.strictEqual(parseSpy.getCall(0).args[0], entries[0]);
+        assert.strictEqual(parseSpy.getCall(1).args[0], entries[1]);
+        done();
+      };
+
+      table.mutate(entries, assert.ifError);
+    });
+  });
+
+  describe('row', function() {
+    var ROW_ID = 'test-row';
+
+    it('should return a Row object', function() {
+      var row = table.row(ROW_ID);
+
+      assert(row instanceof FakeRow);
+      assert.strictEqual(row.calledWith_[0], table);
+      assert.strictEqual(row.calledWith_[1], ROW_ID);
+    });
+  });
+
+  describe('sampleRowKeys', function() {
+    it('should provide the proper request options', function(done) {
+      table.requestStream = function(grpcOpts, reqOpts) {
+        assert.deepEqual(grpcOpts, {
+          service: 'BigtableService',
+          method: 'sampleRowKeys'
+        });
+
+        assert.strictEqual(reqOpts.tableName, TABLE_NAME);
+        assert.strictEqual(reqOpts.objectMode, true);
+        done();
```

Comment:
Pop `done` in a `setImmediate`, so that if the next line causes a blow-up, we know about it.